commit b275946a528d5d85ac063dbe7a60d0cd07708420 (from 182f2cda350b284db5d71cc194ed7f3ea7872d4c)
Merge: 182f2cd 56ff46e
Author: Scott Lahteine <thinkyhead@users.noreply.github.com>
Date:   Fri Apr 24 22:02:37 2015 -0700

    Merge pull request #1966 from thinkyhead/g4_dwell_message
    
    G4 shows status message only if no message is set already

diff --git a/Marlin/Marlin_main.cpp b/Marlin/Marlin_main.cpp
index 65290b3..7f8159c 100644
--- a/Marlin/Marlin_main.cpp
+++ b/Marlin/Marlin_main.cpp
@@ -890,8 +890,11 @@ void get_command() {
 }
 
 bool code_has_value() {
-  char c = strchr_pointer[1];
-  return (c >= '0' && c <= '9') || c == '-' || c == '+' || c == '.';
+  int i = 1;
+  char c = strchr_pointer[i];
+  if (c == '-' || c == '+') c = strchr_pointer[++i];
+  if (c == '.') c = strchr_pointer[++i];
+  return (c >= '0' && c <= '9');
 }
 
 float code_value() {
@@ -1744,14 +1747,15 @@ inline void gcode_G2_G3(bool clockwise) {
 inline void gcode_G4() {
   millis_t codenum = 0;
 
-  LCD_MESSAGEPGM(MSG_DWELL);
-
   if (code_seen('P')) codenum = code_value_long(); // milliseconds to wait
   if (code_seen('S')) codenum = code_value_long() * 1000; // seconds to wait
 
   st_synchronize();
   refresh_cmd_timeout();
   codenum += previous_cmd_ms;  // keep track of when we started waiting
+
+  if (!lcd_hasstatus()) LCD_MESSAGEPGM(MSG_DWELL);
+
   while (millis() < codenum) {
     manage_heater();
     manage_inactivity();

